src/idlcxx/src/streamer_generator.c

```diff
@@ -753,73 +765,82 @@ idl_retcode_t add_null(context_t* ctx, int nbytes, bool stream, bool is_key)
   return IDL_RETCODE_OK;
 }
 
-const char* determine_cast(idl_mask_t mask)
+char* determine_cast(idl_node_t* typespec)
 {
-  if ((mask & IDL_ENUM) == IDL_ENUM)
-    return uint32_cast;
+  if (idl_is_enum(typespec))
+  {
+    char* ns = idl_strdup(""), *returnval = NULL;
+    resolve_namespace(typespec, &ns);
```

Comment:
As a consequence of examining the implementation of resolve_namespace() I discovered something that is not so much an issue in this commit, but in the implementation of the resolve_namespace() function itself. It always uses the IDL name for the namespaces, and forgets to apply the IDL-C++ identifier transformation (i.e. prefix _cxx_ when the identifier clashes with a C++ keyword).

Reply:
You are correct, this was an oversight on my part.
I will fix this.

Reply:
Fix looks good to me.